### PR TITLE
[#46491] handle NC usernames with whitespaces

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/46491

### CODE REVIEW

File: modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_query.rb [hunk 1]
```diff
@@ -58,12 +58,17 @@ def query(parent)
 
     private
 
+    def escape_whitespace(value)
```
Comment (dombesz) [inline modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_query.rb:61]: What do you think of using this sanitizer instead?

```ruby
ActiveStorage::Filename.new("foo/bar.jpg").sanitized
```

https://api.rubyonrails.org/classes/ActiveStorage/Filename.html#method-i-sanitized

Probably there are a few more edge cases than just a whitespace. Also can we test this?

Comment (Kharonus) [reply modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_query.rb:61]: this class is unit tested. I could add a case for the user id containing whitespaces.

Not sure though, about the sanitizer. It actually switches characters, that could be an issue. I'll have a look into it. In general we do not change the values of user id and parent folder, just make them url safe.

Comment (Kharonus) [reply modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_query.rb:61]: Okay, had a look into the sanitizer. It does not solve our issues, as whitespace is no character, that is escaped for file names. As said, we want to escape values (file and folder names, and user ids) for usage in urls here.

**Comment by Kharonus** [comment]
I added an additional unit test to cover that behaviour. I'd suggest, if we find more edge cases, we should add more tests. This code is highly specific for nextcloud. For other storages, there will be other file queries.

Can you have one more look @dombesz?

**Review by dombesz — APPROVED** [review]
